conbench/entities/case.py

```diff
@@ -13,4 +15,24 @@ class Case(Base, EntityMixin):
     tags: Mapped[dict] = NotNull(postgresql.JSONB)
 
 
+def get_case_or_create(case_dict) -> Case:
+    """
+    Try to create case, but expect conflict (work with unique constraint on
+    name/tags).
+
+    Return (newly created, or previously existing) Case object, or raise an
+    exception.
+    """
+    try:
+        return Case.create(case_dict)
+    except s.exc.IntegrityError as exc:
+        if "violates unique constraint" not in str(exc):
+            raise
+
+    Session.rollback()
+    c = Case.first(**case_dict)
+    assert c is not None
```

Comment:
The `assert` statement also enables mypy to infer that this is _not_ None when returning.

Reply:
Nice!